### DISCUSSION TITLE
[Feature] Ability to set original file name

### DISCUSSION THREAD

**Discussion by xobust** [immich-app/immich#12563] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Somtimes the original file name gets lost when procesing it through another application. 
I have personally used Photoprism before and I would like to migrate the original file name to Immich. 

I propose that the update asset api allows submitting a new originalFileName. 
The web & mobile client does not necessarily need to implement this API.

I can help developing this feature. 

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by eikowagenknecht** [comment]
This would indeed be very nice as immich-go (not immich itself) just massively set wrong filenames for my images that I want to manually fix (it removes the "-edited" postfix) but apparently can't without modifying the postgres db myself which I would rather not do.

**Comment by jawgits** [comment]
Yes, being able to edit filenames would be a really useful feature!
